libslide/src/partial_evaluator.rs

```diff
@@ -35,13 +43,13 @@ pub fn evaluate(expr: Stmt, ctxt: EvaluatorContext) -> Result<InternedExpr, Box<
         simplified_expr = flatten_expr(simplified_expr);
     }
     while seen.insert(expr_hash) {
-        for rule in &built_rules {
+        for rule in rules {
             simplified_expr = rule.transform(simplified_expr);
         }
         expr_hash = hash(&simplified_expr);
     }
 
-    Ok(normalize(simplified_expr))
+    normalize(simplified_expr)
```

Comment:
why are we not using results for the evaluating expressions, but we use a result for evaluate. Seems inconsistent

Reply:
because evaluate can fail (see rule_set.build()?) but evaluate_expr will always succeed. Why wrap it in a result if we know it always works?